File: QueDetallitoSSR/Entities/Product/Variant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace QueDetallito.Entities.Product
{
    public class Variant
    {
        public int IdVariant { get; set; }
        public string NameVariant { get; set; }
        public byte Stock { get; set; }
        public Nullable<decimal> Price { get; set; }
        public decimal FinalPrice { get; set; }
        public Nullable<decimal> Discount { get; set; }
        public byte NextAvailability { get; set; }
        public bool CustNumber { get; set; }
        public bool CustLetter { get; set; }
        public bool CustMessage { get; set; }
        public Nullable<int> MessageLength { get; set; }
        public IEnumerable<ProductImages> Images { get; set; }
    }
}

File: QueDetallitoAdmin/ClientApp/src/app/modules/dashboard/components/variant/variant.ts
export class Variant {
    idProduct?: number;
    idVariant?: number;
    nameVariant?: string;
    stock?: number;
    cost?: number;
    price?: number;
    discount?: number;
    nextAvailability?: number;
    custNumber?: boolean;
    custLetter?: boolean;
    custMessage?: boolean;
    active?: boolean;
    messageLength?: number;
}

export class FileImages {
    image?: Array<File>;
}

export class ListVariant {
    idVariant?: number;
    idProduct?: number;
    nameVariant?: string;
    stock?: number;
    cost?: number;
    price?: number;
    discount?: number;
    finalPrice?: number;
    nextAvailability?: number;
    custNumber?: boolean;
    custLetter?: boolean;
    custMessage?: boolean;
    created?: string;
    active?: boolean;
    messageLength?: number;
}

export class ProductImage {
    idProduct?: number;
    idVariant?: number;
    noImage?: number;
    urlLocation?: string;
    created?: string;
}